heutagogy/auth.py

```diff
@@ -1,32 +1,48 @@
 from heutagogy import app
-import flask_login
-
-
-login_manager = flask_login.LoginManager()
-login_manager.init_app(app)
-
-
-class User(flask_login.UserMixin):
-    pass
-
-
-@login_manager.unauthorized_handler
-def unauthorized_handler():
-    return {'error': 'Unauthorized'}, 401
-
+from flask_jwt import JWT
+from werkzeug.security import safe_str_cmp
+
+
+class User(object):
+    def __init__(self, id, username, password):
+        self.id = id
+        self.username = username
+        self.password = password
+
+    def __str__(self):
+        return "User(id='%s')" % self.id
+
+    def __getitem__(self, key):
+        return {
+            'id': self.id,
+            'username': self.username,
+            'password': self.password
+        }.get(key, None)
+
+
+def get_users():
+    users = []
+    for i, x in enumerate(app.config['USERS']):
+        users.append(User(i+1, x['username'], x['password']))
+    return users
+
+
+def authenticate(username, password):
+    users = get_users()
+    username_table = {u.username: u for u in users}
```

Comment:
You traverse the list here. Doing that on each call to the function can be expensive.